src/components/SearchBox.jsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+
+class SearchBox extends React.PureComponent {
+   state = {
+      text: '',
+   }
+
+   handleSubmit = (e) => {
```

Comment:
Try changing this to a function declaration like this:

    handleSubmit(e) {
        e.preventDefault();
        console.log("Searching for:", this.state.text);
    }
You will get a runtime error, read react FAQ to solve it
https://reactjs.org/docs/faq-functions.html#why-is-binding-necessary-at-all